diff --git a/src/lib/common.c b/src/lib/common.c
index aeb74df5..d7d4490c 100644
--- a/src/lib/common.c
+++ b/src/lib/common.c
@@ -352,22 +352,7 @@ int pid_hidepid(void) {
 // print error if unprivileged users can trace the process
 void warn_dumpable(void) {
-	if (getuid() != 0 && prctl(PR_GET_DUMPABLE, 0, 0, 0, 0) == 1 && getenv("FIREJAIL_PLUGIN")) {
-		fprintf(stderr, "Error: dumpable process\n");
-
-		// best effort to provide detailed debug information
-		// cannot use process name, it is just a file descriptor number
-		char path[BUFLEN];
-		ssize_t len = readlink("/proc/self/exe", path, BUFLEN - 1);
-		if (len < 0)
-			return;
-		path[len] = '\0';
-		// path can refer to a sandbox mount namespace, use basename only
-		const char *base = gnu_basename(path);
-
-		struct stat s;
-		if (stat("/proc/self/exe", &s) == 0 && s.st_uid != 0)
-			fprintf(stderr, "Change owner of %s executable to root\n", base);
-		else if (access("/proc/self/exe", R_OK) == 0)
-			fprintf(stderr, "Remove read permission on %s executable\n", base);
+	if (getuid() != 0 && getenv("FIREJAIL_PLUGIN")) {
+		if (prctl(PR_SET_DUMPABLE, 0, 0, 0, 0))
+			errExit("prctl");
 	}
 }
